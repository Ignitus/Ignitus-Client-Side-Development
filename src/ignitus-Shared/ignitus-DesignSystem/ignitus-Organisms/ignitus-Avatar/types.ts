export type avatar = {
  name: string;
  src: string;
};

/** As AvatarImage represents img feel free to add more props here, as <img /> supports lot of other attributes. */
export type AvatarImageProps = {
  src: string;
  alt: string;
};
