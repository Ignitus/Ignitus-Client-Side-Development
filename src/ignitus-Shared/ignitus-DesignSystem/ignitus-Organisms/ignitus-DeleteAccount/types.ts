export interface PropTypes {
  username: string;
}
