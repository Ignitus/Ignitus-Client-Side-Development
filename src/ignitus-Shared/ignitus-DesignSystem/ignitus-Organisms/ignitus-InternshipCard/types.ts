export type CardTypes = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type?: string;
  JobType?: string;
};

export type CardProps = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type: string;
  JobType?: string;
};

export type StyleProps = {
  size?: string;
};
