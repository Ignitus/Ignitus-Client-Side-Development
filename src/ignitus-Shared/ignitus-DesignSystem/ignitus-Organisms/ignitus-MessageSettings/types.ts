export type Props = {
  anyone: boolean;
  connections: boolean;
  directConnections: boolean;
  setAnyone: React.Dispatch<React.SetStateAction<boolean>>;
  setConnections: React.Dispatch<React.SetStateAction<boolean>>;
  setDirectConnections: React.Dispatch<React.SetStateAction<boolean>>;
};
