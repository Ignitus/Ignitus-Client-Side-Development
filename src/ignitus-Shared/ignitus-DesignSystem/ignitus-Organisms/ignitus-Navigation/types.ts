export type displayClassTypes = 'transparent' | 'whitenav';

export interface PureNavigationProps {
  displayClass: displayClassTypes;
  dynamicLogo: any;
};

export interface NavigationProps {
  dynamicNavigation?: boolean;
};
