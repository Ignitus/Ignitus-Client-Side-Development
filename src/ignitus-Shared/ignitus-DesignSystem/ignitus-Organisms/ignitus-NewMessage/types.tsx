export type InputTypes = {
    textColor: string;
    titleColor: string;
    bgColor: string;
  };
  
  export type StyledTagProps = {
    bgColor: string;
    color: string;
  };
  