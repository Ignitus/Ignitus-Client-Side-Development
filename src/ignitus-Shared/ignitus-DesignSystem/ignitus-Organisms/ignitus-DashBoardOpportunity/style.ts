import styled from '@emotion/styled';
import * as T from '../../ignitus-Atoms/typography';
import * as C from '../../ignitus-Atoms/colors';
import { flexibleRowDiv } from '../../shared';
import { Normal, XS } from '../../ignitus-Atoms/fonts';

export const Container = styled.div`
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  border-radius: 20px;
  display: flex;
`;

export const Row = styled(flexibleRowDiv)``;
export const LeftRow = styled.div`
  flex: 1;
  img {
    max-width: 100%;
    max-height: 100%;
    border-top-left-radius: 20px;
    border-bottom-left-radius: 20px;
  }
`;

export const RightRow = styled.div`
  display: flex;
  flex-direction: column;
  flex: 3;
  margin-left: 20px;
  padding: 0.5rem;
`;

export const Description = styled(T.Paragraph)`
  font-weight: ${Normal};
  color: ${C.GreyText};
  margin-right: 1.5rem;
  margin-top: 0.5rem;
`;

export const Title = styled(T.Heading3)`
  flex: 1;
`;

export const SubTitle = styled.p`
  font-weight: ${Normal};
  color: ${C.GreySecondaryText};
  font-size: ${XS};
`;
