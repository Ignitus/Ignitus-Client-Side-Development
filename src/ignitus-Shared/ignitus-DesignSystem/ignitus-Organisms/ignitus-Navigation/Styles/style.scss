$base-color: #000e8c;
$hover-color: #f9bf64;

.navbar {
  align-items: center;
  display:flex;
  position: fixed !important;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 9001;
  justify-content: space-between;
  transition: background-color 300ms ease-in;
}

.navlinks{
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right:1rem;
}

.navlinks-item{
  padding:0rem 1rem;
  align-self: stretch;
}

.navlink{
  color: #fff;
  font-size: 14px;
  font-weight: 700;

  &:hover {
    color: $hover-color;
  }
}

#burger{
  width:1.5rem;
  height:1.5rem;
  position: absolute;
  top:1.2rem;
  right:2rem;
  display:none;
  cursor:pointer;
}

.line{
  display:block;
  background-color: $base-color;
  padding:0.1rem 0.8rem;
  width:0.1rem;
  margin:0.3rem;
  border-radius:1rem;
  transition:transform 100ms ease-in-out;
}

.navbar.whitenav{
  background-color:#fff;
  .navlink{
    color: $base-color;

    &:hover {
      color: $hover-color;
    }
  }
}

.line.mobile:first-child{
  transform: rotateZ(46deg) translateY(0.3rem) translateX(1px);
}

.line.mobile:last-child{
  transform: rotateZ(-43deg) translateY(-0.3rem);
}

.line.mobile:nth-child(2){
  display:none;
}

@media only screen and (max-width:1042px){

  #burger{
    display:block;
    top:1.2rem;
  }

  .navbar{
    flex-direction: column;
    background:#fff;
  }

  .navlinks.mobile{
    display:none;
  }

  .navlinks{
    flex-direction: column;
    align-items:center;
    justify-content: center;
    width:100%;
  }

  .navlinks-item{
    margin-left:0rem;
    margin-bottom:1rem;
  }

  .navbar-brand{
    align-self: flex-start;
    margin-left:5px;
  }

  .navlink{
    color: $base-color;
  }

}
