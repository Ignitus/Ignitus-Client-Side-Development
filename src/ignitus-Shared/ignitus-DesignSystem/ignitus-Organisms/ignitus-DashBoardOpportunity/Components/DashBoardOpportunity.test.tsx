import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';
import { DashBoardOpportunity, Card } from './DashBoardOpportunity';
import { Opportunity } from '../types';
import * as S from '../style';

const props: Opportunity = {
  title: 'Title',
  description: 'Description',
  subtitle: 'subtitle',
  src:
    'https://storage.googleapis.com/ignitus_assets/ig-opportunities/universityOfMelbourne.png',
};

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

it('should rendered', () => {
  const tree = render(<DashBoardOpportunity {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DashBoardOpportunity />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DashBoardOpportunity {...props} />);
  });

  it('should render Card component', () => {
    expect(wrapper.find(Card).length).toEqual(1);
    expect(wrapper.find('img').prop('src')).toEqual(props.src);
    expect(wrapper.find('img').prop('alt')).toEqual(props.title);
    expect(wrapper.find(S.Title).text()).toEqual(props.title);
    expect(wrapper.find(S.SubTitle).text()).toEqual(props.subtitle);
    expect(wrapper.find(S.Description).text()).toEqual(props.description);
  });
});
