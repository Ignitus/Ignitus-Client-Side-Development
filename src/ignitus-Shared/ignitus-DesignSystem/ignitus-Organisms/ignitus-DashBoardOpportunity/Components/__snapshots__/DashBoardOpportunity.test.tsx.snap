// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
.emotion-6 {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
  background-color: #ffffff;
  border-radius: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 img {
  max-width: 100%;
  max-height: 100%;
  border-top-left-radius: 20px;
  border-bottom-left-radius: 20px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
  margin-left: 20px;
  padding: 0.5rem;
}

.emotion-1 {
  font-weight: 500;
  font-size: 22px;
  color: #000066;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-2 {
  font-weight: 400;
  color: #4D4D4D;
  font-size: 14px;
}

.emotion-3 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
  font-weight: 400;
  color: #2D2D2D;
  margin-right: 1.5rem;
  margin-top: 0.5rem;
}

<div
  class="emotion-6"
>
  <div
    class="emotion-5"
  >
    <div
      class="emotion-0"
    >
      <img
        alt="Title"
        src="https://storage.googleapis.com/ignitus_assets/ig-opportunities/universityOfMelbourne.png"
      />
    </div>
    <div
      class="emotion-4"
    >
      <h3
        class="emotion-1"
      >
        Title
      </h3>
      <p
        class="emotion-2"
      >
        subtitle
      </p>
      <p
        class="emotion-3"
      >
        Description
      </p>
    </div>
  </div>
</div>
`;
